kong/plugins/ip-restriction/handler.lua

```diff
@@ -29,6 +33,30 @@ do
 end
 
 
+local is_http_subsystem = ngx.config.subsystem == "http"
+
+
+local do_exit
+if is_http_subsystem then
+  do_exit = function(status, message)
+    return kong.response.error(status, message)
+  end
+
+else
+  do_exit = function(status, message)
+    local tcpsock, err = ngx_req.socket(true)
+    if err then
+      error(err)
+    end
+
+    tcpsock:send(cjson_encode({
+      message = message
+    }))
+
+    return ngx_exit(status)
```

Comment:
Could this just use `kong.response.exit`?

```suggestion
    kong.response.exit(status, { message = message })
```

Reply:
Can kong.response.exit(status) work with the TCP connection?

Reply:
Yes, see `pdk/response.lua`

Reply:
Seems like this is not addressed.

Reply:
The Kong response caused an error when I tested it.
However, I'll remove the json response that should make it more efficient.

Reply:
I removed the json response.

Reply:
@hanshuebner is this acceptable?

Reply:
I think @bungle pointed out that `kong.response.exit` should be preferred to `ngx.exit`.  Any reason not to follow his advice?

Reply:
Yes, I tried using kong.response.exit and it failed sending the reponse.

Reply:
@hanshuebner can we close this?  I went with one of @bungle suggestions.  The other didn't work.

Reply:
It's not sending the message.  I'm seeing an empty string:
https://github.com/scrudge/kong/actions/runs/5391318477/jobs/9787991403#step:9:1342